content/en/docs/reference/glossary/mesh-federation.md

```diff
@@ -0,0 +1,7 @@
+---
+title: Mesh Federation
+---
+A loosely-coupled set of autonomous [service meshes](#service-mesh) collectively providing the services
+for an application. Two service meshes in a federation are isolated from each other and have different service
```

Comment:
I think we may need to work on this definition a bit...

- what does `isolated` mean here?
- does mesh federation require independent administrative control? Or simply different domains of trust?
- it seems that we need to talk more directly about the two meshes selectively exposing services to each other for consumption. And to do that, they have to share identities and trust so that workloads across both meshes can mutually authenticate with each other.

Also, have we officially accepted the definition of `Mesh Federation`? I feel like there is still some internal debate, but I could be wrong.

Reply:
I'm trying to use this definition as a starting point to get something officially accepted :)

Maybe saying `isolated` is being too restrictive? Actually, I think mesh federation doesn't "require" independent administrative control or even different domains of trust. Instead I think it should just say "clusters can have different administrative control / domains of trust".

I'm just trying to give the high level (what is it) definition here. Things like how the federation is achieved (expose-consume) would be described in the concepts (not the glossary). It's a baby step.

Reply:
Understood ... really what you're grappling with here is that age old question: "what is a mesh?". I had to go through that same exercise when writing the concept docs.

I think we should drop the concept of administrative domain completely and focus on service naming and trust domain, and leave it at that. BTW, if we introduce the term "trust domain", we'll need to define that too :).

Reply:
I find the term "administrative domain" as a synonym for mesh to be quite useful because it approaches the definition of mesh boundary from a users perspective; mesh federation is the mechanism by which services in meshes under the administration of distinct entities can interoperate. 

What CA cert the administrator chooses to use for a mesh (the trust domain) is just one of many choices an administrator would make for a mesh. I do not see these terms as synonymous.

Reply:
@shalako

I'm not against adding the term "administrative domain" so long as we have a clear, concise definition for it. In fact we probably need it.

Assuming its inclusion, a mesh would consist of:

- exactly 1 administrative domain
- exactly 1 trust domain
- common naming

The administrative domain is a little different since it's more of an abstract concept and doesn't have any concrete representation per se. However, I think it's true to say that the boundary of a mesh represents a single administrative domain. In that sense, an administrative domain's concrete representation is the mesh itself.

@smawson @louiscryan thoughts?

Reply:
I tried to explain the single administrative domain concept in application terms. Not sure it was clear enough but I was trying say that it essentially combines a set of services into a composite application under a single control. Take a look at the first paragraph [here](https://istio.io/docs/concepts/multicluster-deployments/). Does that seem reasonable?

Reply:
@frankbu A few things...

- We probably want to steer away from the term `microservices` and instead just use `services`, which is something we have a definition for.
- I don't think it's necessarily true to say that a mesh is a single composite application. It could be multiple. If a single mesh is used by a company, for example, they may offer multiple composite application within the same mesh, using namespace isolation between teams.

Reply:
Good point about the single application wording. Combining services into logical applications is the real value of service mesh, even if it's not limited to just one. I still wonder when there is more than one logical application in the same mesh, if it's still sort of one big application from the administrators point of view, even if the administration of different tenants, for example, are delegated to others. Any idea what would be a better word to use to describe this centrally administered set of one or more logical applications?

Reply:
I think there are a couple distinct administration roles here:

- mesh administrator: responsible for the installation and management of the control plane throughout the mesh.
- tenant administrator: responsible for the deployment of services to a single namespace within the mesh.

These may be the same person/group or they may not. In large orgs, they will likely be different.

It *may* be true that each tenant deploys a single composite application. I guess I'm not sure I see the need for calling out (and defining) composite applications at all.

Reply:
I tend to agree with you. The only thing I'm still struggling with is whether, and how, we should associate applications with the general term service mesh. It seems to me that the main reason service meshes have come about is to tie back together the services that implement an application after moving from a monolith to microservices-based architecture. That's the purpose of a service mesh that I believe most people understand, so it seems that we should somehow be mentioning applications in the definition of service mesh, even if they aren't necessarily 1-1.

Reply:
In an attempt to help craft these definitions, I would like to ask that we write them following the following established pattern:
Start with the term (word or phrase) to be defined.
Followed by the class of object or concept to which the term belongs.
Finish with the differentiating characteristics that distinguish it from all others of its class.

https://owl.purdue.edu/owl/general_writing/common_writing_assignments/definitions.html

Applying the pattern above to the proposed definition I come up with the following suggestion:

```
A mesh federation is a set of autonomous meshes that are loosely-coupled to collectively provide the services of an application. An autonomous service mesh is a mesh that fulfills the following requirements:

- Isolated from other meshes
- Has different service definitions than other meshes
- Has an independent administrative control
```

Missing from my suggestion is the explanation for `loosely-coupled` and it should be added IMO.